### PR TITLE
docs(HRKA-0000): update docs with redirect error codes and add basic …

### PR DESCRIPTION
**Pull request by danielgratzl**
…documentation for testing services

### CODE REVIEW

File: TestingYourService.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+| 4 | Aktualisieren der Berechtigungen abgebrochen    | <ul><li>Benutzer wird auf `/authorization/update` mit den entsprechenden Parametern geleitet.</li><li>User bricht ab</li></ul>                   | Redirect an Service `redirect_uri` mit `error=cancelled` Parameter                                                                 |
+| 5 | Verbindung trennen                              | <ul><li>Benutzer wird auf `/authorization/revoke` mit den entsprechenden Parametern geleitet.</li><li>User entzieht berechtigungen</li></ul>     | Redirect an Service `redirect_uri`                                                                                                 |
+| 6 | Verbindung trennen abgebrochen                  | <ul><li>Benutzer wird auf `/authorization/revoke` mit den entsprechenden Parametern geleitet.</li><li>User bricht ab</li></ul>                   | Redirect an Service `redirect_uri` mit `error=cancelled` Parameter                                                                 |
```
Comment (joel-bach) [inline TestingYourService.md:46]: I think what I am missing in general is a note regarding the api-configuration endpoint as the source of truth regarding permissions. I'd also add them to the test cases as part of the "Ablauf" to make sure the data project is aware that this is the way to do it.

**Review by joel-bach — APPROVED** [review]
Good addition 👌 

There are a couple of commas/spelling things I'd change but a) I am not sure how important it is and b) probably is easier to not do these as single comments. What do you think?

**Comment by danielgratzl** [comment]
@joel-bach I wouldn't mind if you fixed smaller typos etc. during the review. If there are a lot I might need to check whether the spellchecker in IntellJ works as intended though 🤔

File: README.md [hunk 2]
```diff
@@ -161,15 +182,18 @@ sequenceDiagram
 
 ## API Verwendung
 
-Nachdem der Service den Authorisierungsvorgang für eine Praxis erfolgreich abgeschlossen und damit ein gültiges Accesstoken erhalten hat können Requests gegen die API ausgeführt werden.
+Nachdem der Service den Autorisierungsvorgang für eine Praxis erfolgreich abgeschlossen und damit ein gültiges Access-Token erhalten hat können Requests gegen die API ausgeführt werden.
 
 Die Peer-to-peer Architektur von Heureka ermöglicht den direkten und sicheren Zugriff auf die Daten des Praxisinformationssystems in dem alle Anfragen via eines Proxy-Setups direkt auf das Primärsystem gelangen.
 
+### <a name="api_configuration"></a>API Konfiguration
```
Comment (joel-bach) [inline README.md:189]: FYI there are automated anchors in Github, in this case it would be #api-konfiguration

File: README.md [hunk 3]
```diff
@@ -161,15 +182,18 @@ sequenceDiagram
...
+
+Der API-Konfiguration Endpunkt liefert zusammen mit der HCP-spezifischen URL für die FHIR API auch die Proxykonfiguration zurück, welche für Requests gegen die FHIR-API verwendet werden muss.
+Ausserdem enthält die API-Konfiguration die aktuell für die Installation gültigen Berechtigungen (Grants).
 
-### API Konfiguration
+Dieser Endpunkt liefert immer die **aktuell gültigen** Berechtigungen für das verwendete Token zurück. Er sollte dazu verwendet werden, um jederzeit sicherzustellen, dass der Service die erforderlichen Berechtigungen besitzt.
 
-Der API-Konfigurationsendpunkt liefert zusammen mit der HCP-spezifischen URL für die FHIR API auch die Proxykonfiguration zurück, welche für Requests gegen die FHIR-API verwendet werden muss.
-Ausserdem enthält die API-Konfiguration die aktuell für die Installation gültigen Berechigungen (Grants).
+Als Best Practice wird empfohlen, die gültigen Berechtigungen vor der ersten User-Interaktion zu überprüfen, um damit sicherzustellen, dass nachfolgende API Zugriffe erlaubt sind.
```
Comment (joel-bach) [inline README.md:196]: Some suggestions:

```suggestion
Der API-Konfigurations-Endpunkt liefert zusammen mit der HCP-spezifischen URL für die FHIR-API auch die Proxykonfiguration zurück, welche für Requests gegen die FHIR-API verwendet werden muss.
Ausserdem enthält die API-Konfiguration die aktuell für die Installation gültigen Berechtigungen (Grants).

Dieser Endpunkt liefert immer die **aktuell gültigen** Berechtigungen für das verwendete Token zurück. Er sollte dazu verwendet werden, um sicherzustellen, dass der Service die erforderlichen Berechtigungen besitzt.

Als Best Practice wird empfohlen, die gültigen Berechtigungen vor der ersten User-Interaktion zu überprüfen, um damit sicherzustellen, dass nachfolgende API-Zugriffe erlaubt sind. Auch sollte nach dem Ende des Grant-/Update-/Revoke-Flows der aktuelle Stand der Berechtigungen anhand des Endpunkts überprüft werden.

Hinweis: Es ist in seltenen Fällen möglich, dass eine Differenz zwischen dem Resultat des API-Konfigurations-Endpunkt und den Berechtigungen bei der Abfrage von Daten besteht, da die Berechtigungen nicht unmittelbar synchronisiert werden. Die "Source of Truth" ist in dem Fall immer der API-Konfigurations-Endpunkt und nach spätestens 60 Sekunden sollten die Systeme wieder eine einheitliche Datengrundlage haben.
```

File: TestingYourService.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+| 7 | Benutzer aktualisiert Berechtigungen via Portal | <ul><li>Benutzer verwendet Portal um zusätzliche Berechtigungen zu vergeben oder zu entziehen.</li></ul>                                         | Neue Berechtigungen werden nach kurzer Zeit (max. 60 Sekunden) aktiv und können via API-Konfiguration Endpunkt abgerufen werden.   |
+| 8 | Benutzer trennt Verbindung via Portal           | <ul><li>Benutzer verwendet Portal um die Verbindung vom Service zu trennen.</li></ul>                                                            | Refresh Token verliert Gültigkeit und Berechtigungen für aktives Access Token werden nach kurzer Zeit (max. 60 Sekunden) entzogen. |
```
Comment (joel-bach) [inline TestingYourService.md:49]: Some suggestions (mostly regarding consistency):

```suggestion
# Testszenarien für Services

Um User:innen ein möglichst gutes Erlebnis rund um die Verwendung von Heureka zu bieten, ist es sinnvoll, verschiedene Szenarien zu berücksichtigen und zu testen.
Dies betrifft vor allem die Autorisierungs-Flows, die sowohl aus dem Heureka Portal als auch serviceseitig initiiert werden können.
Diese sind nachfolgend beschrieben.

Um die aktuellen Berechtigungen zu überprüfen, muss der (API-Konfigurations-Endpunkt)[./README.md#api_configuration] verwendet werden.

## Initiale Freigabe

Die initiale Freigabe der Praxis für den Service erfolgt immer serviceseitig, indem der/die Benutzer:in auf das Portal geleitet wird, um den Authorization Code Flow zu starten. 

## Trennen der Verbindung

### Heureka-Portal

Der Service muss die Verbindungstrennung entsprechend erkennen und kann beispielsweise die Möglichkeit bieten, sich erneut mit Heureka zu verbinden.

**Preview:** Wird die Verbindung via Heureka-Portal getrennt, wird der Service darüber via Webhook informiert, wenn eine Webhook-URL konfiguriert ist.

### Service

Wird das Trennen der Verbindung vom Service aus initiiert, so wird anschliessend wieder an die angegebene `redirect_uri` zurück zum Service geleitet. Der Service muss anschliessend (mithilfe des API-Konfigurations-Endpunkts) überprüfen, ob die Verbindung erfolgreich getrennt wurde.

## Berechtigungen aktualisieren

Wenn ein Service mit optionalen Berechtigungen registriert wurde, hat der/die Heureka Benutzer:in jederzeit die Möglichkeit, dem Service diese Berechtigungen zu gewähren oder zu entziehen. 

### Heureka-Portal

Der Service muss damit umgehen können, dass sich die vergebenen Berechtigungen ändern können (mithilfe des API-Konfigurations-Endpunkts können die aktuellen Berechtigungen für die mit dem Access-Token verbundene Installation abgefragt werden).

**Preview:** Werden die Berechtigungen via Heureka-Portal aktualisiert, wird der Service darüber via Webhook informiert, wenn eine Webhook URL konfiguriert ist.

### Service

Wird eine Aktualisierung der Berechtigungen vom Service aus initiiert, so wird der Benutzer anschliessend wieder via der angegeben `redirect_uri` zurück zum Service geleitet. Der Service muss anschliessend (mithilfe des API-Konfigurations-Endpunkts) überprüfen, ob die Berechtigungen wie erwartet angepasst wurden.

## Testprotokoll
| # | Beschreibung                                    | Ablauf                                                                                                                                           | Erwartetes Resultat                                                                                                                |
|---|-------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| 1 | Zugriff erfolgreich beantragen                  | <ul><li>Benutzer wird auf `/authorization/grant` mit den entsprechenden Parametern geleitet.</li><li>User autorisiert Service.</li></ul>          | Redirect an Service `redirect_uri` mit `code`-Parameter um Access- & Refreshtoken zu lösen                                         |
| 2 | Zugriffsantrag abgebrochen                      | <ul><li>Benutzer wird auf `/authorization/grant` mit den entsprechenden Parametern geleitet.</li><li>User bricht ab.</li></ul>                    | Redirect an Service `redirect_uri` mit `error=access_denied`-Parameter                                                             |
| 3 | Berechtigungen aktualisieren                    | <ul><li>Benutzer wird auf `/authorization/update` mit den entsprechenden Parametern geleitet.</li><li>User aktualisiert Berechtigungen.</li></ul> | Redirect an Service `redirect_uri`                                                                                                 |
| 4 | Aktualisieren der Berechtigungen abgebrochen    | <ul><li>Benutzer wird auf `/authorization/update` mit den entsprechenden Parametern geleitet.</li><li>User bricht ab.</li></ul>                   | Redirect an Service `redirect_uri` mit `error=cancelled`-Parameter                                                                 |
| 5 | Verbindung trennen                              | <ul><li>Benutzer wird auf `/authorization/revoke` mit den entsprechenden Parametern geleitet.</li><li>User entzieht berechtigungen.</li></ul>     | Redirect an Service `redirect_uri`                                                                                                 |
| 6 | Verbindung trennen abgebrochen                  | <ul><li>Benutzer wird auf `/authorization/revoke` mit den entsprechenden Parametern geleitet.</li><li>User bricht ab.</li></ul>                   | Redirect an Service `redirect_uri` mit `error=cancelled`-Parameter                                                                 |
| 7 | Benutzer aktualisiert Berechtigungen via Portal | <ul><li>Benutzer verwendet Portal um zusätzliche Berechtigungen zu vergeben oder zu entziehen.</li></ul>                                         | Neue Berechtigungen werden nach kurzer Zeit (max. 60 Sekunden) aktiv und können via API-Konfigurations-Endpunkt abgerufen werden.   |
| 8 | Benutzer trennt Verbindung via Portal           | <ul><li>Benutzer verwendet Portal um die Verbindung vom Service zu trennen.</li></ul>                                                            | Refreshtoken verliert Gültigkeit und Berechtigungen für aktives Accesstoken werden nach kurzer Zeit (max. 60 Sekunden) entzogen. |
```

**Review by joel-bach — COMMENTED** [review]
I've added some suggestions regarding typos/punctuation. Nothing critical IMO but might be nice to increase consistency.

**Pull request merged**